### PR TITLE
CreatePostScreen implementation & Test Suite

### LINKED ISSUE #64: Create the create feed screen

### PR DESCRIPTION
**Pull request by Atomium90**
**Description**
Implements the screen allowing users to create a new post with the following inputs:
- **Title**: single-line text field
- **Body**: multi-line field with fixed height and auto-scroll to avoid oversized blocks
- **Tags**: free-form tags added via input, normalized and displayed in a scrollable container limited to 3 visible rows

**Navigation**
- Integrated into the app's navigation flow
- `onPost`, `onDiscard`, and `onBack` callbacks are wired and functional

**Test Suite**
- Adds a complete test suite for `CreatePostScreen`
- Covers all major flows: input, tag lifecycle, validation, post creation, error handling, and navigation
- Optimized for density over quantity: fewer tests, each covering multiple behaviors to reduce CI overhead

**Notes**
- Tag normalization includes hash prefix and case-insensitive deduplication
- UI auto-scrolls on body and tag overflow for better UX
- Test suite follows Meeple Meet's global architecture and was initially generated with Claude 4.5, then manually reviewed and refined

_This PR was reformulated with Copilot to improve readability._

### CODE REVIEW

File: app/src/main/java/com/github/meeplemeet/ui/CreatePostScreen.kt [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+  val bodyScroll = rememberScrollState()
+  var tagInput by remember { mutableStateOf("") }
+  val selectedTags = remember { mutableStateListOf<String>() }
+
+  var isPosting by remember { mutableStateOf(false) }
+  val snackbarHostState = remember { SnackbarHostState() }
+  val scope = rememberCoroutineScope()
+  val focusManager = LocalFocusManager.current
+
+  fun normalizeTag(input: String): String {
```
Comment (NilSamaha) [inline app/src/main/java/com/github/meeplemeet/ui/CreatePostScreen.kt:95]: add kotlin doc for this function

File: app/src/main/java/com/github/meeplemeet/ui/CreatePostScreen.kt [hunk 2]
```diff
@@ -0,0 +1,317 @@
...
+    val t = input.trim()
+    if (t.isEmpty()) return ""
+
+    val withoutHashes = t.trimStart('#').trim()
+    if (withoutHashes.isEmpty()) return ""
+    return "#" + withoutHashes.lowercase()
+  }
+
+  fun addTag() {
```
Comment (NilSamaha) [inline app/src/main/java/com/github/meeplemeet/ui/CreatePostScreen.kt:108]: add kotlin doc for this function

**Review by NilSamaha — APPROVED** [review]
#### 1. Code Structure & Readability
The screen is well-implemented but quite long. Consider breaking CreatePostScreen into smaller composables, to improve readability and maintainability.
#### 2. Helper Functions
Good use of helper functions like normalizeTag and addTag. Adding Kotlin docs would make their behavior and expected inputs/outputs clearer for future contributors.
#### 3. UI Behavior & Auto-Scrolling
The LaunchedEffect auto-scroll for the body and tags is clever. Consider extracting it into a helper composable or function to reduce duplication and simplify maintenance.
#### 4. Testing & TestTags
Test tags are consistently applied and make the UI tests clear and reliable. Excellent attention to test coverage and edge cases.

Overall Feedback:
Great work, the code is clean, the UI behavior is well thought out, and the tests are thorough. Small improvements in composable separation, documentation, and helper extraction would further improve maintainability.
